rct229/rules/section6/section6rule9.py

```diff
@@ -0,0 +1,135 @@
+                schedules_p = data["schedules_p"]
+                return {
+                    **data,
+                    "building_open_schedule_p":  getattr_(
+                        find_one_with_field_value("$", "id", building_p["building_open_schedule"], schedules_p),
+                        "schedule", "hourly_values")
+                }
+
+            class ZoneRule(RuleDefinitionListIndexedBase):
+                def __init__(self):
+                    super(Section6Rule9.RuleSetModelInstanceRule.BuildingRule.ZoneRule, self).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, True),
+                        each_rule=Section6Rule9.RuleSetModelInstanceRule.BuildingRule.ZoneRule.SpaceRule(),
+                        index_rmr="proposed",
+                        list_path="spaces[*]",
+                    )
+
+                def create_data(self, context, data=None):
+                    zone_p = context.proposed
+                    return {
+                        **data,
+                        "avg_space_height": zone_p.get("volume", 0.0) / pint_sum(
+                            find_all("spaces[*].floor_area", zone_p), ZERO.AREA)
+                    }
+
+                def list_filter(self, context_item, data=None):
+                    # filter out spaces with no interior lighting
+                    space_b = context_item.baseline
+                    return space_b.get("interior_lighting", None)
+
+                class SpaceRule(RuleDefinitionBase):
+                    def __init__(self):
+                        super(Section6Rule9.RuleSetModelInstanceRule.BuildingRule.ZoneRule.SpaceRule, self).__init__(
+                            rmrs_used=UserBaselineProposedVals(False, True, True)
+                        )
+
+                    def get_calc_vals(self, context, data=None):
+                        space_p = context.proposed
+                        space_b = context.baseline
+                        avg_space_height = data["avg_space_height"]
+                        schedules_b = data["schedules_b"]
+                        schedules_p = data["schedules_p"]
+                        building_open_schedule_p = data["building_open_schedule_p"]
+
+                        normalized_schedule_b = normalize_interior_lighting_schedules(space_b, avg_space_height, schedules_b)
+                        normalized_schedule_p = normalize_interior_lighting_schedules(space_p, avg_space_height, schedules_p)
+                        schedule_comparison_result_dictionary = compare_schedules(normalized_schedule_p, normalized_schedule_b,
+                                                                       building_open_schedule_p, 1.0)
+                        daylight_control = functools.reduce(lambda daylight_control_a, daylight_control_b: daylight_control_a | daylight_control_b, find_all("interior_lighting[*].are_schedules_used_for_modeling_daylighting_control", space_p))
```

Comment:
I appreciate the use of functools.  I am always looking for good uses for it.
In this case, however, I think the logic could be simplified considerably:
```
daylight_control = any(
    find_all(
        "interior_lighting[*].are_schedules_used_for_modeling_daylighting_control", 
        space_p
    )
)
```

Reply:
Great to offer this elegant and simple solution. I never knew this built-in function before. Thanks!